.github/workflows/ci.yml

```
@@ -17,14 +17,20 @@ jobs:
   # Only then, normal testing proceeds
   unit-tests:
     needs: qa
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: [3.8, 3.9, 3.10, 3.11]
```

Need to be in commas so 3.10 is understood correctly and not as 3.1.
```suggestion
        python-version: ["3.8", "3.9", "3.10", "3.11"]
```